bin/main.asm

```diff
@@ -40,13 +40,16 @@ trap:
 
 	mov cx, TRAP_SIZE
 	rep movsb
+	sub di, TRAP_SIZE
 	
 	push ds
```

Comment:
Yes, because earlier in the loop we need to write to ES (as opposed to reading from it) so we have to swap somewhere. There is an alternative: Finish writing the trap with a placeholder value and then replacing the placeholder with the correct value. This has a disadvantage because if another player is hit, they are likely to run the placeholder value before we can manage to replace it. The other way is to split the movsb into two parts and in between use a stosb. The second option takes slightly more space and may or may not be faster (one less useless write, but using rep for big writes is faster than for many smaller writes).